packages/typespec-ts/src/modular/buildCodeModel.ts

```diff
@@ -1644,7 +1629,7 @@ export function emitCodeModel(
   simpleTypesMap.clear();
   const allModels = getAllModels(dpgContext);
```

Comment:
[iscai-msft](https://github.com/iscai-msft)

Reply:
Since `Pet` is not a discriminated type, `Dog` and `Cat` will not be used in `read`, so `getAllModel` will not return these two orphan models.

Reply:
Thanks for the info! I am checking the semantic with TypeSpec team: https://github.com/microsoft/typespec/issues/2450.